### PR TITLE
Add optional role config for Snowflake connectors

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

To allow users to connect using a different role, in case the default one doesn't work.

### 🤓 What?

As titled. Also simplify the `connect` method.

### 🧪 Tested?

Locally tested with & without the `role` config set.

### CODE REVIEW

**Review by alyiwang — APPROVED** [review]
LGTM

File: metaphor/snowflake/auth.py [hunk 1]
```diff
@@ -89,17 +77,16 @@ def connect(config: SnowflakeAuthConfig) -> snowflake.connector.SnowflakeConnect
...
-            },
-        )
-
-    raise ValueError(
-        "Invalid Snowflake configuration, please set either password or private key"
+    return snowflake.connector.connect(
+        account=config.account,
+        user=config.user,
+        password=config.password,
+        private_key=pkb,
```
Comment (alyiwang) [inline metaphor/snowflake/auth.py:84]: I just thought of an edge case: if a user provided both password and key, what will happen? Maybe we should add this to the validator too.

Comment (mars-lan) [reply metaphor/snowflake/auth.py:84]: Good point. Internally it must prefer one over the other but we should fail loudly in that case. Will create a follow-up PR.

Comment (mars-lan) [reply metaphor/snowflake/auth.py:84]: https://github.com/MetaphorData/connectors/pull/296

**Pull request merged**